src/datepicker/datepicker-input.ts

```diff
@@ -410,6 +410,16 @@ export class NgbInputDatepicker implements OnChanges,
   ngOnChanges(changes: SimpleChanges) {
     if (changes['minDate'] || changes['maxDate']) {
       this._validatorChange();
+
+      if (this.isOpen()) {
+        if (changes['minDate']) {
+          this._cRef.instance.minDate = this._dateAdapter.toModel(changes.minDate.currentValue);
+        }
+        if (changes['maxDate']) {
+          this._cRef.instance.maxDate = this._dateAdapter.toModel(changes.maxDate.currentValue);
+        }
+        this._cRef.instance.ngOnChanges(changes);
```

Comment:
```ts
this._cRef.instance.ngOnChanges(changes);
```
I believe this bit is not required,  or at least should not directly use `NgOnChanges` and be changed to use of CDR:
```ts
this._cRef.changeDetectorRef.detectChanges();
```

Reply:
Dynamic components loading will not trigger `ngOnChanges` in child component even after calling `detectChanges` because this event is reserved only for template bindings. 
Here, `datepicker-input` dynamically uses `datepicker` component and hence this will not trigger ngOnChanges. So, It has to be done manually. 
This is the reason behind the above case.